api_keys/utils.py

```diff
@@ -9,12 +10,27 @@
 def redis_connection():
     global _connection
     if _connection is None:
-        _connection = redis.StrictRedis(
-            host=settings.REDIS_DB_HOST,
-            port=settings.REDIS_DB_PORT,
-            db=settings.REDIS_DB_NUMBER,
-            password=settings.REDIS_DB_PASSWORD
-        )
+        if settings.REDIS_SENTINEL_HOSTS is not None:
+            # If we have listed any sentinels, use those to manage the connection.
+            # REDIS_SENTINEL_HOSTS will be a dict, but this needs an array of tuples [('host', port)]
+            sentinel = Sentinel(
+                [(host, port) for host, port in settings.REDIS_SENTINEL_HOSTS.items()],
+                socket_timeout=0.1
+            )
+            _connection = sentinel.master_for(
+                settings.REDIS_SENTINEL_SET,
+                socket_timeout=0.1,
+                db=settings.REDIS_DB_NUMBER,
+                password=settings.REDIS_DB_PASSWORD
+            )
+        else:
+            # Otherwise fall back to a regular connection
+            _connection = redis.StrictRedis(
```

Comment:
Would be good to have a test that overrides REDIS_SENTINEL_HOSTS to None to test this fall back - is that possible though, or not because it's global, hmm